### ISSUE TITLE
IMAP list folder, encoding error ?

### ISSUE THREAD

**Issue by YZJClear** [karastojko/mailio#132]
vcpkg: mailio[core]:x64-windows-static -> 0.21.0#2

code:
`mailio::imap conn("xxx", 143);

conn.authenticate("admin", "xxx", mailio::imap::auth_method_t::LOGIN);
mailio::imap::mailbox_folder_t fld = conn.list_folders("");
print_folders(0, fld);`

**this result is cant reader, my inbox has chinese char**
result:
`INBOX
        &V4NXPg-
        AZ
                AZ_&TgtTVQ-
                AZ_&U9GNJw-
                AZ_&UKxrPg-
                AZ_&XAFT9w-
                AZ_&Y9CNJw-
                AZ_&e35lNg-
                AZ_&e35lNngB-
                BS_&UWVekw-
                BS_&VAhedg-
                BS_&Y9CNJw-
                SF_&UWVekw-
                SF_&Y9CNJw-
                ZZ_&UWVekw-
        E-invoice
        ECS
                        &WSl,,E6R-
        EM
                                MAIL
                                VFEMAIL
        HKA
        HKA_&U9GNJw-`

**Comment by tsurumi-yizhou** [comment]
This just deal with the transferring but not content of character coding. Problem of which should be your reader.

**Comment by karastojko** [comment]
So, your folders on the IMAP server have non-Latin characters and listing them throws an error?

**Comment by YZJClear** [comment]
> So, your folders on the IMAP server have non-Latin characters and listing them throws an error?

Yes, the server contains simplified Chinese and traditional Chinese, they can be displayed normally, javax.mail. It also works

**Comment by karastojko** [comment]
It is the feature I have never tested. Let me try it. Maybe I can easily add it as a new feature.

**Comment by YZJClear** [comment]
Looking forward to your updates

**Comment by YZJClear** [comment]
![image](https://github.com/karastojko/mailio/assets/34211642/ca630586-50f6-4932-919f-bf34ade0ac29)
It doesn't look like any coding, how should i handle that it should be Chinese characters

**Comment by tsurumi-yizhou** [comment]
The character coding should be automatically  recognized by your reader from MIME.

**Comment by YZJClear** [comment]
> The character coding should be automatically recognized by your reader from MIME.

你知道怎么解决吗

**Comment by karastojko** [comment]
IMAP folders may be UTF7 encoded which is not currently supported by *mailio*. Thus, it is not a bug but the missing feature.

**Comment by YZJClear** [comment]
> IMAP folders may be UTF7 encoded which is not currently supported by _mailio_. Thus, it is not a bug but the missing feature.

Hello, I have two questions. First, do you have plans to add this feature? Second, are you Chinese? (I'm Chinease)

**Comment by karastojko** [comment]
I would have to check for a UTF-7 encoding C++ library, does it exist with MIT license or I should create such utility. Also, I need to see whether UTF-8 is also being used for this purposes. So, right now I do not have the straight-forward answer to your question.
Regarding the nationality - I am not Chinese.

**Comment by YZJClear** [comment]
Thank you, I will try the method you mentioned in the future.
